src/main/java/io/github/linuxforhealth/core/expression/condition/CompoundAndOrCondition.java

```diff
@@ -0,0 +1,81 @@
+package io.github.linuxforhealth.core.expression.condition;
+
+import com.google.common.base.Preconditions;
+import io.github.linuxforhealth.api.Condition;
+import io.github.linuxforhealth.api.EvaluationResult;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import java.text.BreakIterator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CompoundAndOrCondition implements Condition {
+    private final String conditionStatement;
+
+    public CompoundAndOrCondition(String conditionStatement) {
+        this.conditionStatement = conditionStatement;
+    }
+
+    @Override
+    public boolean test(Map<String, EvaluationResult> contextVariables) {
+        return Boolean.parseBoolean(makeRecursive(conditionStatement, contextVariables));
+    }
+
+    private String makeRecursive(String conditionStatementContainer, Map<String, EvaluationResult> contextVariables) {
+        // will not handle nested parens
+        Pattern pattern = Pattern.compile("\\(.*?\\)");
+        Matcher matcher = pattern.matcher(conditionStatementContainer);
+
+        while (matcher.find()) {
+            String group = matcher.group();
+            conditionStatementContainer = eval(conditionStatementContainer, group, contextVariables);
+        }
+
+        while(conditionStatementContainer.contains("&&") || conditionStatementContainer.contains("||")) {
+            if(conditionStatementContainer.contains("&&") && conditionStatementContainer.contains("||") && !conditionStatementContainer.contains("(")) {
+                conditionStatementContainer = fixParens(conditionStatementContainer);
+                conditionStatementContainer = makeRecursive(conditionStatementContainer, contextVariables);
+            } else {
+                conditionStatementContainer = eval(conditionStatementContainer, conditionStatementContainer, contextVariables);
+            }
+        }
+
+        return conditionStatementContainer;
+    }
+
+    private String fixParens(String conditionStatement) {
```

Comment:
I'm having a bit of a hard time following what's happening with this function, could we add some documentation? It sort of seems like this is trying to deterministically handle a situation where there are still `&&` and `||` but no more parens and the code will favor  grouping the or?

Also looking at the coverage report it looks like only the `else` branch is executed